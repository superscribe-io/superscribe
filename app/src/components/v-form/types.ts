import { DeepPartial, Field } from '@directus/types';

export type FormField = DeepPartial<Field> & {
	field: string;
	name: string;
	hideLabel?: boolean;
	hideLoader?: boolean;
};
