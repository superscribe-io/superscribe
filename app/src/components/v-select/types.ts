export type Option = {
	value: string | number | null;
	icon?: string;
	text?: string;
	disabled?: boolean;
	children?: Option[];
	divider?: boolean;
	selectable?: boolean;
	hidden?: boolean;
};
