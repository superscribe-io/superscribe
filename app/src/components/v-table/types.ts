export type Alignment = 'left' | 'center' | 'right';

export type HeaderRaw = {
	text: string;
	value: string;
	description?: string | null;
	align?: Alignment;
	sortable?: boolean;
	width?: number | null;
	[key: string]: any;
};

export type Header = Required<HeaderRaw>;

export type Item = {
	[key: string]: any;
};

export type ItemSelectEvent = {
	value: boolean;
	item: Item;
};

export type Sort = {
	by: string | null;
	desc: boolean;
};
