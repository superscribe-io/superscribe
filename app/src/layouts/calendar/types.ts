export type LayoutOptions = {
	template?: string;
	startDateField?: string;
	endDateField?: string;
	firstDay?: number;
	viewInfo?: {
		type: string;
		startDateStr: string;
	};
};
