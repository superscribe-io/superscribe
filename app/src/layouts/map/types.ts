import { CameraOptions } from 'maplibre-gl';

export type LayoutQuery = {
	fields: string[];
	sort: string[];
	limit: number;
	page: number;
};

export type LayoutOptions = {
	cameraOptions?: CameraOptions & { bbox: any };
	geometryField?: string;
	autoLocationFilter?: boolean;
	clusterData?: boolean;
	animateOptions?: any;
	displayTemplate?: string;
};
