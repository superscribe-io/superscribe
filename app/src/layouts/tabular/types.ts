export type LayoutOptions = {
	widths?: {
		[field: string]: number;
	};
	align?: {
		[field: string]: 'left' | 'center' | 'right';
	};
	limit?: number;
	spacing?: 'comfortable' | 'cozy' | 'compact';
};

export type LayoutQuery = {
	fields: string[];
	sort: string[];
	page: number;
	limit: number;
};
