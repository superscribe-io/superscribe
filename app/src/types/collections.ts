import { Collection as CollectionRaw, CollectionType } from '@superscribe/types';
import { TranslateResult } from 'vue-i18n';

export interface Collection extends CollectionRaw {
	name: string | TranslateResult;
	icon: string;
	type: CollectionType;
	color?: string | null;
}
