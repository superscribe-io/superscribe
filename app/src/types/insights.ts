export type Dashboard = {
	id: string;
	name: string;
	note: string;
	icon: string;
	color: string;
	date_created: string;
	user_created: string;
};
