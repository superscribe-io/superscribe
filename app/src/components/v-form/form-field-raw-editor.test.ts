import { it, test, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';

import formFieldRawEditor from './form-field-raw-editor.vue';
import { GlobalMountOptions } from '@vue/test-utils/dist/types';

const i18n = createI18n({ legacy: false });

const global: GlobalMountOptions = {
	plugins: [i18n],
};

test('should render', () => {
	expect(formFieldRawEditor).toBeTruthy();

	const wrapper = mount(formFieldRawEditor, {
		props: {
			showModal: true,
			field: 'object',
			disabled: false,
			currentValue: '["id","new_content"]',
		},
		global,
	});

	expect(wrapper.html()).toMatchSnapshot();
});

// test if there is a value
test('submitting', async () => {
	expect(formFieldRawEditor).toBeTruthy();

	const wrapper = mount(formFieldRawEditor, {
		props: {
			showModal: true,
			field: 'string',
			disabled: false,
			currentValue: 'things',
		},
		global,
	});

	const button = wrapper.findAll('v-button').at(1);
	await button!.trigger('click');
	await wrapper.vm.$nextTick();
	expect(wrapper.emitted().setRawValue.length).toBe(1);
});

it('should cancel with keydown', async () => {
	const wrapper = mount(formFieldRawEditor, {
		props: {
			showModal: true,
			field: 'object',
			disabled: false,
			currentValue: '["id","new_content"]',
		},
		global,
	});

	await wrapper.trigger('esc');
	await wrapper.vm.$nextTick();
	expect(wrapper.emitted().cancel.length).toBe(1);
});

it('should cancel with the cancel button', async () => {
	const wrapper = mount(formFieldRawEditor, {
		props: {
			showModal: true,
			field: 'object',
			disabled: false,
			currentValue: '["id","new_content"]',
		},
		global,
	});

	const button = wrapper.findAll('v-button').at(0);
	await button!.trigger('click');
	await wrapper.vm.$nextTick();
	expect(wrapper.emitted().cancel.length).toBe(1);
});
