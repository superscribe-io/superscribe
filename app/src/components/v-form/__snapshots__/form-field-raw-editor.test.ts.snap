// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render 1`] = `
"<v-dialog data-v-33a4ea83=\\"\\" model-value=\\"true\\" persistent=\\"\\">
  <v-card data-v-33a4ea83=\\"\\">
    <v-card-title data-v-33a4ea83=\\"\\">edit_raw_value</v-card-title>
    <v-card-text data-v-33a4ea83=\\"\\">
      <interface-system-raw-editor data-v-33a4ea83=\\"\\" type=\\"undefined\\" disabled=\\"false\\" language=\\"plaintext\\" placeholder=\\"enter_raw_value\\"></interface-system-raw-editor>
    </v-card-text>
    <v-card-actions data-v-33a4ea83=\\"\\">
      <v-button data-v-33a4ea83=\\"\\" secondary=\\"\\">cancel</v-button>
      <v-button data-v-33a4ea83=\\"\\">done</v-button>
    </v-card-actions>
  </v-card>
</v-dialog>"
`;
